jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -309,6 +299,30 @@ def _warn_if_no_equation(self):
         else:
             return False
 
+    def _update_viewer_filters(self, event={}):
+        if event.get('new', self.cube_fit):
+            # only want image viewers in the options
+            self.add_results.viewer.filters = ['is_image_viewer']
+        else:
+            # only want spectral viewers in the options
+            self.add_results.viewer.filters = ['is_spectrum_viewer']
+
+    @observe('cube_fit')
+    def _cube_fit_changed(self, event={}):
+        self._update_viewer_filters(event=event)
+
+        if event.get('new'):
+            self._units['y'] = self.app._get_display_unit('sb')
+            self.dataset.add_filter('is_flux_cube')
+            self.dataset.remove_filter('layer_in_spectrum_viewer')
+        else:
+            self._units['y'] = self.app._get_display_unit('spectral_y')
+            self.dataset.add_filter('layer_in_spectrum_viewer')
+            self.dataset.remove_filter('is_flux_cube')
+
+        self.dataset._clear_cache()
+        self.reestimate_model_parameters()
```

Comment:
If we remove `reestimate_model_parameters` from here does this still work (just sometimes requiring the user to hit the button)?

I wonder if its ever useful to keep the estimates from the non cube fit as the starting point (especially if there is no unit conflict)?  Or if the user manually entered there own values but then they were changed by toggling cube fit, would that be unexpected/disruptive?

Reply:
Updated to only reestimate if `spectral_y` and `sb` have different units.